Assets/Scripts/AI/Base/AIManager.cs

```diff
@@ -1,225 +1,396 @@
-    }
-
-    public void SwitchCombatState(bool enabled)
-    {
-        isInCombat = enabled;
-        animationManager.animator.SetBool(animationManager.animations.isInCombat, isInCombat);
-    }
-
-     public void SetAttackCooldown(float time)
-    {
-        attackCooldownTimestamp = Time.time + time;
-    }
-
-    public void ResetAttackCooldown()
-    {
-        attackCooldownTimestamp = 0;
-    }
-
-    public bool IsCooldownExpired()
-    {
-        return Time.time > attackCooldownTimestamp;
-    }
-
-    public bool IsCombatEnabled()
-    {
-        return isInCombat;
-    }
-
-    public void StopMovement()
-    {
-        navMeshAgent.velocity = Vector3.zero;
-        navMeshAgent.speed = 0f;
-        navMeshAgent.isStopped = true;
-    }
-
-    public bool Dodge(bool dodge)
-    {
-        return animationManager.Dodge(dodge);
-    }
-
-    public Vector3 GetTargetPosition()
-    {
-        if(chaseTarget == null)
-        {
-            Debug.LogError("Target not found.");
-            return transform.position;
-        }
-        return new Vector3(chaseTarget.position.x, transform.position.y, chaseTarget.position.z);
-    }
-
-    public bool IsNavMeshAgentMoving()
-    {
-        return !navMeshAgent.isStopped;
-    }
-
-    public void MoveNavMeshAgent(Vector3 destination, float speed)
-    {
-        navMeshAgent.destination = destination;
-		navMeshAgent.speed = speed;
-		navMeshAgent.isStopped = false;
-    }
-
-    public bool IsTimestampSet()
-    {
-        return isTimestampSet;
-    }
-
-    public void SetTimestamp(float seconds)
-    {
-        timestamp = Time.time + seconds;
-        isTimestampSet = true;
-    }
-
-    public bool IsTimestampExpired(){
-        bool expired = timestamp < Time.time;
-        if(expired)
-        {
-            isTimestampSet = false;
-        }
-        return expired && isTimestampSet;
-    }
-
-    public void TakeDamage(float amount)
-    {
-        currentHealth -= amount;
-        if(currentHealth <= 0)
-        {
-            StopMovement();
-            SwitchCombatState(false);
-            controller.Die();
-            animationManager.Die();
-        }
-    }
-     public void EnableMarkers()
-    {
-        markerManager.EnableMarkers(unarmedCombatSettings.unarmedAttackDamage);
-    }
-
-    public void DisableMarkers()
-    {
-        markerManager.DisableMarkers();
-    }
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+using UnityEngine.AI;
+
+[RequireComponent(typeof(AIAnimationManager))]
+public class AIManager : MonoBehaviour
+{
+    public List<Transform> wayPointList;
+    public GameObject eyes;
+    public AIStats aiStats;
+
+    [HideInInspector] public NavMeshAgent navMeshAgent;
+    [HideInInspector] public int nextWayPoint;
+    [HideInInspector] public AIAnimationManager animationManager;
+    [HideInInspector] public Vector3 lastKnownTargetPosition;
+    [HideInInspector] public Vector3 walkTarget;

```

Comment:
Why are we hiding components and attributes in the script from the asset manager?